trio/_channel.py

```diff
@@ -0,0 +1,279 @@
+from collections import deque, OrderedDict
+from math import inf
+
+import attr
+from outcome import Error, Value
+
+from . import _core
+from .abc import SendChannel, ReceiveChannel
+
+# rename SendChannel/ReceiveChannel to SendHandle/ReceiveHandle?
+# eh, maybe not -- SendStream/ReceiveStream don't work like that.
+
+# send or send_object? eh just send is good enough
+
+# implementing this interface on top of a stream is very natural... just
+# pickle/unpickle (+ some framing). Actually, even clone() is not bad at
+# all... you just need a shared counter of open handles, and then close the
+# underlying stream when all SendChannels are closed.
+
+# to think about later:
+# - max_buffer_size=0 default?
```

Comment:
I'd support this - it seems to go hand-in-hand with the docs saying "if in doubt, use 0".